### ISSUE TITLE
Do not use bitsize-dependent values in the GnoVM

### ISSUE THREAD

**Issue by thehowl** [gnolang/gno#3288]
Like `math.MinInt`.

I realised while reviewing some code that `values_conversions` uses `math.MinInt` / `math.MaxInt` for float-integer conversion. This value will change if the GnoVM is compiled for 32-bits. We should find other similar values that depend on the integer size, and replace them to always have the value of an integer within the gnovm at 64-bits, independently of the underlying machine architecture.

**Comment by petar-dambovaliev** [comment]
I would like to hear more people's opinions. My immediate reaction is that maybe we shouldn't target 20 year old 32bit systems. Most libraries and software, including operating systems are removing that support. Why do we want to add it?

**Comment by mvertes** [comment]
For the sake of simplicity, I think that we should not attempt to support 32 bits architectures. Gno is not meant for low-power embedded systems.

**Comment by mvertes** [comment]
i.e. It's not the code itself, it's the cost of support. I do not see us to double the test, CI and maintenance efforts to support 32 bits systems (only armv7 relevant now).

**Comment by thehowl** [comment]
yes, agreed; but we should still use 64-bit values everywhere under the hood.

**Comment by odeke-em** [comment]
A way to simplify and codify this decision is to make a file inside perhaps the gnovm/pkg/gnolang directory `nocompile_386.go` or add the build tag in the file `//go:build 386`

```go
//go:build 386

package whatever

func init() {
   panic("Gno doesn't run on 32-bit machines")
}
```

this will ensure that 32-bit machine compilation will always fail without having to go make enforcements in the Gnovm which could take forever.

**Comment by odeke-em** [comment]
Since PR https://github.com/gnolang/gno/pull/3643 is merged, what other things are left to do here? Can we perhaps close this issue?

**Comment by thehowl** [comment]
Let's close this, though I think we should also then merge #3591
